flask_resources/deserializers/json.py

```diff
@@ -17,4 +17,4 @@ class JSONDeserializer(DeserializerMixin):
 
     def deserialize_data(self, data, *args, **kwargs):
         """Deserializes the input data into json."""
-        return json.loads(data)
+        return json.loads(data) if data else {}
```

Comment:
Doubt: In which cases would the `else` clause happen?

Reply:
For any request with no payload. For DELETE for instance. See the test I've added in the PR: https://github.com/inveniosoftware/flask-resources/pull/46/files#diff-c24f9f74a93cc9e07203da5e9e0522c3R70 . In this case, `json.loads(data)` would try to load an empty bytestream (empty `request.data`) and error out even though passing an empty payload is fine.

Reply:
For requests wihout a payload. For instance, for DELETE call. See the test I've added in the PR: https://github.com/inveniosoftware/flask-resources/pull/46/files#diff-c24f9f74a93cc9e07203da5e9e0522c3R70 . Without this check, `json.loads(data)` tries to load an empty bytestream (`request.data` is empty) and errors out even though an empty payload is fine.